### PR TITLE
gh-99955: standardize return values of functions in compiler's code-gen

### PR DESCRIPTION
**Pull request by iritkatriel**
Consistently return -1 for error, 0 for success. 

* Issue: gh-99955

### CODE REVIEW

File: Python/compile.c [hunk 1]
```diff
@@ -55,6 +55,14 @@
  */
 #define STACK_USE_GUIDELINE 30
 
+#define SUCCESS 0
+#define ERROR -1
+
+#define RETURN_IF_ERROR(X)  \
+    if ((X) == -1) {        \
+        return -1;          \
```
Comment (markshannon) [inline Python/compile.c]:
```suggestion
        return ERROR;          \
```

Perhaps

**Review by markshannon — APPROVED** [review]
Thanks for doing this. This makes the code so much easier to follow and maintain.

**Comment by markshannon** [comment]
Actually, it is probably best to add a news item, so anyone familiar with the code knows about the change.

File: Python/compile.c [hunk 2]
```diff
@@ -6888,8 +6810,8 @@ compiler_pattern_or(struct compiler *c, pattern_ty p, pattern_context *pc)
     // Need to NULL this for the PyObject_Free call in the error block.
     old_pc.fail_pop = NULL;
     // No match. Pop the remaining copy of the subject and fail:
-    if (!cfg_builder_addop_noarg(CFG_BUILDER(c), POP_TOP, LOC(p)) ||
-        !jump_to_fail_pop(c, LOC(p), pc, JUMP)) {
+    if ((cfg_builder_addop_noarg(CFG_BUILDER(c), POP_TOP, LOC(p)) < 0) ||
```
Comment (iritkatriel) [inline Python/compile.c]:
```suggestion
    if (cfg_builder_addop_noarg(CFG_BUILDER(c), POP_TOP, LOC(p)) < 0 ||
```

File: Python/compile.c [hunk 3]
```diff
@@ -55,6 +55,14 @@
  */
 #define STACK_USE_GUIDELINE 30
 
+#define SUCCESS 0
+#define ERROR -1
+
+#define RETURN_IF_ERROR(X)  \
+    if ((X) == -1) {        \
+        return ERROR;          \
```
Comment (iritkatriel) [inline Python/compile.c]:
```suggestion
    if ((X) == -1) {        \
        return ERROR;       \
```

**Pull request merged**